application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,165 @@
+    private static final Color AMBIENT_COLOR = Color.black;
+    private final Cache<Long, Instant> authorIdToLastCommandInvocation = createCooldownCache();
+    private final Predicate<String> modMailChannelNamePredicate;
+
+
+    /**
+     * Creates a new instance.
+     *
+     * @param jda the JDA instance to use to retrieve guildCache
+     * @param config the config to use for this
+     */
+    public ModMailCommand(JDA jda, Config config) {
+        super(COMMAND_NAME, "Send a message to the moderators of the selected guild",
+                CommandVisibility.GLOBAL);
+
+        OptionData messageOption = new OptionData(OptionType.STRING, OPTION_MESSAGE,
+                "What do you want to tell them?", true);
+        OptionData guildOption = new OptionData(OptionType.STRING, OPTION_SERVER_GUILD,
+                "Your server guild ID", true);
+        OptionData anonymousOption = new OptionData(OptionType.BOOLEAN, OPTION_STAY_ANONYMOUS,
+                "if set, your name is hidden", true);
+
+        List<Command.Choice> choices = jda.getGuildCache()
+            .stream()
+            .map(guild -> new Command.Choice(guild.getName(), guild.getIdLong()))
+            .toList();
+
+        guildOption.addChoices(choices);
+
+        getData().addOptions(messageOption, guildOption, anonymousOption);
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        long userId = event.getUser().getIdLong();
+        event.deferReply().setEphemeral(true).queue();
+
+        if (isChannelOnCooldown(userId)) {
+            event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+        authorIdToLastCommandInvocation.put(userId, Instant.now());
+
+        Optional<TextChannel> modMailAuditLog = getChannel(event);
+        if (modMailAuditLog.isEmpty()) {
+            logger.warn("Cannot find the designated modmail channel in this server.");
+            return;
+        }
+
+        String userMessage = event.getOption(OPTION_MESSAGE).getAsString();
+        boolean wantsToStayAnonymous = event.getOption(OPTION_STAY_ANONYMOUS).getAsBoolean();
+        String user = event.getUser().getAsMention();
+        MessageCreateAction message = modMailAuditLog.orElseThrow()
+            .sendMessageEmbeds(createModMailMessage(user, userMessage))
+            .addActionRow(DiscordClientAction.General.USER.asLinkButton("Click to see profile!",
```

Comment:
profile of whom? maybe just "Author profile".

also, what if the user wanted to stay anonymous? then this button shouldnt be there.

its confusing that u first create the message like this and then, later, overwrite it (if anonymous). do some if-else instead. also, use more helper methods to separate the parts of the command. ur `onSlashCommand` is way too long and does too many things. it should be sepaarated into 4 sections:
* cooldown
* find channel
* create message
* send message etc

and those should be moved into helper methods, such that the 4 section become very visibile when reading `onSlashCommand`